### ISSUE TITLE
ChakraCore crashes due to another null dereference

### ISSUE THREAD

**Issue by leeswimming** [microsoft/ChakraCore#5944]
The release build of ChakraCore crashes due to null dereferencing.

PoC:
```
function func_0() {                                                                
    try {                                                                          
        var var_0 = arguments;                                                     
        throw new Error();                                                         
    } catch (undefined) {                                                          
        func_0(typeof var_0 === 1);                                                
    }                                                                              
}                                                                                  
func_0();
```

Build Option: None
OS: Ubuntu 18.04.1 LTS
Arch: x86_64

**Comment by akroshg** [comment]
I can confirm it repros with debug build.
I get 
ASSERTION 12176: (c:\cg\core\lib\runtime\Language/JavascriptOperators.inl, line 20) GetTypeId aValue is null
 Failure: (aValue != nullptr)
It does not repro nonative. It seems we bailout in the try/catch block.

**Comment by MikeHolman** [comment]
Fixed by #6150
